### PR TITLE
feat!: add cloud logging option for public DNS

### LINKED ISSUE #46: Enable cloud logging for public DNS zones
### TL;DR

Since Google Provider v.4.40 creates the functionality to use terraform to activate Logging on Managed DNS zones, it would be nice to have this feature in the module. Same as https://github.com/terraform-google-modules/terraform-google-cloud-dns/issues/35 

### Terraform Resources

```markdown
resource "google_dns_managed_zone" "cloud-logging-enabled-zone" {
  name        = "cloud-logging-enabled-zone"
  dns_name    = "services.example.com."
  description = "Example cloud logging enabled DNS zone"
  labels = {
    foo = "bar"
  }

  cloud_logging_config {
    enable_logging = true
  }
}

https://registry.terraform.io/providers/hashicorp/google/latest/docs/resources/dns_managed_zone
```

### Detailed design

_No response_

### Additional information

_No response_

### PR DESCRIPTION
**Pull request by imrannayer**
fix #46 #35

### CODE REVIEW

**Comment by comment-bot-dev** [comment]
@imrannayer
Thanks for the PR! 🚀<br/>✅ Lint checks have passed.

**Comment by comment-bot-dev** [comment]
@imrannayer
Thanks for the PR! 🚀<br/>✅ Lint checks have passed.

**Comment by imrannayer** [comment]
@bharathkkb @g-awmalik can u plz review?
Thanks

**Comment by comment-bot-dev** [comment]
@imrannayer
Thanks for the PR! 🚀<br/>✅ Lint checks have passed.

**Review by bharathkkb — COMMENTED** [review]
Thanks @imrannayer 
Could we bump min versions to 4.40 per https://github.com/hashicorp/terraform-provider-google/blob/main/CHANGELOG.md#4420-october-31-2022

File: examples/public-zone/variables.tf [hunk 1]
```diff
@@ -42,3 +42,9 @@ variable "labels" {
     version = "bar"
   }
 }
+
+variable "enable_logging" {
```
Comment (bharathkkb) [inline examples/public-zone/variables.tf:46]: nit: lets not add this here and just hard code in example if we want to test this field. We are trying to move towards having most inputs to the module inlined within the example.

Comment (imrannayer) [reply examples/public-zone/variables.tf]: done.

**Comment by comment-bot-dev** [comment]
@imrannayer
Thanks for the PR! 🚀<br/>✅ Lint checks have passed.

**Review by bharathkkb — APPROVED** [review]

**Pull request merged**